### ISSUE TITLE
from vllm._C import cuda_utils raise error

### ISSUE THREAD

**Issue by WanBenLe** [vllm-project/vllm#2797]
ImportError: /root/autodl-tmp/conda/envs/wslconda/lib/python3.9/site-packages/vllm/_C.cpython-39-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops15to_dtype_layout4callERKNS_6TensorEN3c108optionalINS5_10ScalarTypeEEENS6_INS5_6LayoutEEENS6_INS5_6DeviceEEENS6_IbEEbbNS6_INS5_12MemoryFormatEEE

Used these command in linux.

# Install vLLM with CUDA 11.8.
export VLLM_VERSION=0.2.4
export PYTHON_VERSION=39
pip install https://github.com/vllm-project/vllm/releases/download/v${VLLM_VERSION}/vllm-${VLLM_VERSION}+cu118-cp${PYTHON_VERSION}-cp${PYTHON_VERSION}-manylinux1_x86_64.whl

# Re-install PyTorch with CUDA 11.8.
pip uninstall torch -y
pip install torch --upgrade --index-url https://download.pytorch.org/whl/cu118

# Re-install xFormers with CUDA 11.8.
pip uninstall xformers -y
pip install --upgrade xformers --index-url https://download.pytorch.org/whl/cu118

nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0

Maybe the torch and xFormers version should be set?

**Comment by subiawaud** [comment]
Same issue for me, following similar instructions.

**Comment by 1860yk** [comment]
same issue.

**Comment by ohaijen** [comment]
exact same issue, same steps to reproduce.

**Comment by samuelbacaner** [comment]
I have the same issue as well on Python 3.10.

**Comment by santhnm2** [comment]
Same issue on Python 3.11.

**Comment by LiuXiaoxuanPKU** [comment]
Hi folks, I can reproduce the error. It seems that the wheel will install torch 2.2.0 and xformers 0.0.24 by default. And torch 2.2.0 is causing the error here.

If you downgrade xformers to 0.0.23 by `pip install xformers==0.0.23`, it will downgrade torch to 2.1.1 as well and should solve the problem.

**Comment by LiuXiaoxuanPKU** [comment]
Please let me know if  downgrading xformers still does not work. It fixes the error on my side.

**Comment by santhnm2** [comment]
This series of commands worked for me:
```
export VLLM_VERSION=0.2.6
export PYTHON_VERSION=311
pip install https://github.com/vllm-project/vllm/releases/download/v${VLLM_VERSION}/vllm-${VLLM_VERSION}+cu118-cp${PYTHON_VERSION}-cp${PYTHON_VERSION}-manylinux1_x86_64.whl

# install
pip uninstall torch -y
pip install torch==2.1.2 --index-url https://download.pytorch.org/whl/cu118

# Install xFormers
pip uninstall xformers -y
pip install xformers==0.0.23 --index-url https://download.pytorch.org/whl/cu118
```

**Comment by LiuXiaoxuanPKU** [comment]
> This series of commands worked for me:
> 
> ```
> export VLLM_VERSION=0.2.6
> export PYTHON_VERSION=311
> pip install https://github.com/vllm-project/vllm/releases/download/v${VLLM_VERSION}/vllm-${VLLM_VERSION}+cu118-cp${PYTHON_VERSION}-cp${PYTHON_VERSION}-manylinux1_x86_64.whl
> 
> pip uninstall torch -y
> pip install torch==2.1.2 --index-url https://download.pytorch.org/whl/cu118
> 
> pip uninstall xformers -y
> pip install xformers==0.0.23 --index-url https://download.pytorch.org/whl/cu118
> ```
> 
> However, I do see this pip error:
> 
> ```
> ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
> vllm 0.2.6+cu118 requires torch==2.1.2, but you have torch 2.1.1+cu118 which is incompatible.
> vllm 0.2.6+cu118 requires xformers==0.0.23.post1, but you have xformers 0.0.23+cu118 which is incompatible.
> ```

Yeah xformer 0.0.23 here I am doing it specifically for vllm 0.2.4.

For vllm 0.2.6, it requires torch 2.1.2, xformers == 0.0.23.post1 as shown [here](https://github.com/vllm-project/vllm/blob/671af2b1c0b3ed6d856d37c21a561cc429a10701/requirements.txt#L8).

The point here is that torch 2.2.0 is causing the problem, you need to downgrade torch to a version that your version of vllm requires.

**Comment by santhnm2** [comment]
> > This series of commands worked for me:
> > ```
> > export VLLM_VERSION=0.2.6
> > export PYTHON_VERSION=311
> > pip install https://github.com/vllm-project/vllm/releases/download/v${VLLM_VERSION}/vllm-${VLLM_VERSION}+cu118-cp${PYTHON_VERSION}-cp${PYTHON_VERSION}-manylinux1_x86_64.whl
> > 
> > pip uninstall torch -y
> > pip install torch==2.1.2 --index-url https://download.pytorch.org/whl/cu118
> > 
> > pip uninstall xformers -y
> > pip install xformers==0.0.23 --index-url https://download.pytorch.org/whl/cu118
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > However, I do see this pip error:
> > ```
> > ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
> > vllm 0.2.6+cu118 requires torch==2.1.2, but you have torch 2.1.1+cu118 which is incompatible.
> > vllm 0.2.6+cu118 requires xformers==0.0.23.post1, but you have xformers 0.0.23+cu118 which is incompatible.
> > ```
> 
> Yeah xformer 0.0.23 here I am doing it specifically for vllm 0.2.4.
> 
> For vllm 0.2.6, it requires torch 2.1.2, xformers == 0.0.23.post1 as shown [here](https://github.com/vllm-project/vllm/blob/671af2b1c0b3ed6d856d37c21a561cc429a10701/requirements.txt#L8).
> 
> The point here is that torch 2.2.0 is causing the problem, you need to downgrade torch to a version that your version of vllm requires.

Got it, thanks - I updated my commands to this and now everything seems to be working smoothly:
```
export VLLM_VERSION=0.2.6
export PYTHON_VERSION=311
pip install https://github.com/vllm-project/vllm/releases/download/v${VLLM_VERSION}/vllm-${VLLM_VERSION}+cu118-cp${PYTHON_VERSION}-cp${PYTHON_VERSION}-manylinux1_x86_64.whl

# install
pip uninstall torch -y
pip install torch==2.1.2 --index-url https://download.pytorch.org/whl/cu118

pip uninstall xformers -y
pip install xformers==0.0.23.post1 --index-url https://download.pytorch.org/whl/cu118
```

**Comment by NikolaBorisov** [comment]
I'm seeing this issue with the master branch. I can not build working docker image.

**Comment by manoj2312** [comment]
I solved it by installing vllm==0.3.1 with cu12.2 and pytorch==2.1.2. vLLM is super fast especially batching, its worth it. Following are some dependencies it installed:
**Dependencies:**
aioprometheus-23.12.0 cupy-cuda12x-12.1.0 fastapi-0.109.2 fastrlock-0.8.2 httptools-0.6.1 msgpack-1.0.7 ninja-1.11.1.1 orjson-3.9.14 pynvml-11.5.0 quantile-python-1.1 ray-2.9.2 starlette-0.36.3 transformers-4.37.2 uvicorn-0.27.1 uvloop-0.19.0 vllm-0.3.1 watchfiles-0.21.0 websockets-12.0 xformers-0.0.23.post1

**Comment by surak** [comment]
This is not completed at all. The pypi package is broken.
